### PR TITLE
Rewrite button press animation

### PR DESCRIPTION
**Pull request by osdnk**

### CODE REVIEW

File: src/components/animations/ButtonPressAnimation.js [hunk 1]
```diff
@@ -68,280 +47,282 @@ const HapticFeedbackTypes = {
...
+  const gestureState = useAnimatedValue(0);
+  const scaleValue = useAnimatedValue(1);
+  const finished = useAnimatedValue(0);
+  const frameTime = useAnimatedValue(0);
+  const time = useAnimatedValue(0);
+  const toValue = useAnimatedValue(0.5);
+  const durationVal = useAnimatedValue(duration);
+  const animationState = useAnimatedValue(3);
+
+  const inteactionHandle = React.useRef();
```
Comment (jinchung) [inline src/components/animations/ButtonPressAnimation.js]: Missing "r"; rename to "interactionHandle"

File: src/components/animations/ButtonPressAnimation.js [hunk 2]
```diff
@@ -68,280 +47,282 @@ const HapticFeedbackTypes = {
...
+  const createHandle = React.useCallback(() => {
+    inteactionHandle.current = InteractionManager.createInteractionHandle();
+  });
```
Comment (jinchung) [inline src/components/animations/ButtonPressAnimation.js]: Can you include dependency list to all useCallbacks in this file (empty [] or with proper dependencies)

File: src/components/animations/ButtonPressAnimation.js [hunk 3]
```diff
@@ -1,47 +1,34 @@
-import { omit, pick } from 'lodash';
 import PropTypes from 'prop-types';
-import React, { Fragment, Component } from 'react';
+import React from 'react';
```
Comment (jinchung) [inline src/components/animations/ButtonPressAnimation.js]: Could you import { useCallback, useEffect, useMemo, useRef, useState }

**Comment by christianbaroni** [comment]
@osdnk @mikedemarais this PR needs to be reconciled with these changes: https://github.com/rainbow-me/rainbow/commit/c99d2141ae2de912bf6c6292c5097dde6e0f6e01#diff-3adf3d613ca026f7d63ca87fd57e8835

Currently this PR is importing transformOrigin from ./procs, which no longer exists.

And on develop, ButtonPressAnimation seems to be broken (I tapped the screen only twice in the beginning): https://gfycat.com/slipperybelovedivorygull

@osdnk probably best to catch up with develop and reassess.

**Comment by christianbaroni** [comment]
One last thing @osdnk, can we add support for `scaleTo` > `1`?

Right now it causes a crash, "invariant violation, input range must be monotonically non-decreasing"

**Comment by christianbaroni** [comment]
Found another @osdnk. Pressing a contact doesn't do anything. Long press works fine though.

https://gfycat.com/faithfulfreedavidstiger

**Comment by osdnk** [comment]
I believe I managed to fix all issues

**Pull request merged**